### PR TITLE
add test coverage and update documentation

### LINKED ISSUE #33: add "HOWTO test the code(unit test and e2e test)"
We need a document for anyone wanting to either write code which directly accesses the Cyclone API, or to contribute directly to the Cyclone project. It should include:

- [x] API doc(Just move the  API doc to this document)
- [x] The process of developing and contributing code to the Cyclone project. (Same as https://github.com/kubernetes/kubernetes/blob/master/docs/devel/README.md)
- [x] HOWTO setup dev environment
- [ ] HOWTO test the code(unit test and e2e test)

### PR DESCRIPTION
**Pull request by gaocegege**
fixes #33 #42

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL

**Comment by gaocegege** [comment]
@coveralls thank you, baby :)

File: docs/developer-guide.md [hunk 1]
```diff
@@ -20,6 +20,32 @@ The script would run all the dependencies in docker containers, compile and run
...
+
+### Unit Testing
+
+You could run unit test by typing:
+
+```shell
+./tests/run-unit-test.sh
+```
+
+And Our travis CI jobs would also run these, so you could also check out the build log to get the test result in travis. In the future we would run these tests in cyclone.
```
Comment (superxi911) [inline docs/developer-guide.md]: we would run these tests by cyclone

File: docs/developer-guide.md [hunk 2]
```diff
@@ -20,6 +20,32 @@ The script would run all the dependencies in docker containers, compile and run
...
+
+Now cyclone's e2e test would start a local circle server and register local docker host as a worker to circle server. Then it would start a standalone process to run the e2e test cases by sending RESTful requests.
+
+You could run e2e test by typing:
+
+```shell
+./tests/run-e2e.sh
+```
+
+In the future we would run these tests in cyclone.
```
Comment (superxi911) [inline docs/developer-guide.md]: by cyclone

**Comment by gaocegege** [comment]
PTAL again @superxi911

**Comment by superxi911** [comment]
LGTM

**Pull request merged**